ui/src/app/job-details/common/failures-table/failures-table.component.ts

```diff
@@ -12,13 +12,15 @@ export class JobFailuresTableComponent implements OnInit {
   @Input() failures: FailureMessage[];
   @Input() showHeaders: boolean;
   @Input() numToShow: number;
-  displayedColumns: string[] = ['name', 'message', 'links'];
+  @Input() displayedColumns: string[];
   dataSource: FailureMessage[] | null;
 
   constructor() {}
 
   ngOnInit() {
-    this.dataSource = this.failures.slice(0, this.numToShow);
+    if (this.failures) {
+      this.dataSource = this.failures.slice(0, this.numToShow);
```

Comment:
It doesn't seem like you should be leaving dataSource `null`, e.g. `hasTaskNames()` L38